lib/knigge/implementation.ex

```diff
@@ -9,11 +9,13 @@ defmodule Knigge.Implementation do
 
   alias Knigge.Options
 
-  def fetch!(%Options{implementation: {:config, otp_app, key}, default: default}) do
-    if is_nil(default) do
-      Application.fetch_env!(otp_app, key)
-    else
-      Application.get_env(otp_app, key, default)
+  def fetch!(%Options{implementation: {:config, otp_app, [key | keys]}, default: default}) do
+    with nil <- otp_app |> env!(key, default) |> get(keys) do
+      raise ArgumentError,
+        message: """
+        could not fetch application environment #{inspect([key | keys])} \
+        for application #{inspect(otp_app)}\
+        """
```

Comment:
If it's okay I'd prefer to write it like this:

```suggestion
  def fetch!(%Options{implementation: {:config, otp_app, [key | keys]}, default: default}) do
    otp_app
    |> env!(key, default)
    |> get(keys)
    |> case do
      nil ->
        raise ArgumentError,
          "could not fetch application environment #{inspect([key | keys])} for application #{inspect(otp_app)}"
        
      implementation ->
        implementation
    end
```